modules/seqexec/web/client/src/main/scala/seqexec/web/client/actions.scala

```diff
@@ -195,9 +196,19 @@ object actions {
   // Used for UI debugging
   final case class MarkStepAsRunning(s: Observation.Id, step: Int) extends Action
 
-  private val stepInfo: PartialFunction[Step, (StepId, StepState, List[(Resource, ActionStatus)])] = {
+  // This, together with the whole pprint library, is shaken off when linking
+  // for production, since show is not called in that case (see LoggingProcessor).
+  private object Printer {
+    private val pprinter: PPrinter =
+      PPrinter(defaultHeight = 200, colorApplyPrefix = fansi.Color.Blue)
+
+    def apply(x: Any): String =
+      fansi.Str.join(pprinter.tokenize(x, initialOffset = 4).toSeq:_*).toString
```

Comment:
`fansi`?

Reply:
It's Haoyi's ANSI coloring library.

Reply:
In theory, we could just call `pprinter.apply` here, which does the same as this `.tokenize` call but drops the `initialOffset` parameter (seemed like a bug in pprint).

However, even this way the `initialOffset` parameter is not being honoured, or it does not do what I expect it to do (which is shift the whole block by that number of spaces). It still looks much better than whithout `pprint` though.

I guess that I could simplify back to `pprinter.apply` and either report/submit a PR to `pprint`.